mobile/mobile_test.go

```diff
@@ -38,6 +45,50 @@ func TestWrapper_StartAgain(t *testing.T) {
 	}
 }
 
+func TestWrapper_SignUpWithEmail(t *testing.T) {
+	_, ref, err := test.CreateReferral(test.RefKey, 1)
+	if err != nil {
+		t.Errorf("create referral for signup failed: %s", err)
+		return
+	}
+	if len(ref.RefCodes) == 0 {
+		t.Error("create referral for signup got no codes")
+		return
+	}
+	err = wrapper.SignUpWithEmail(cusername, cpassword, cemail, ref.RefCodes[0])
+	if err != nil {
+		t.Errorf("signup failed: %s", err)
+		return
+	}
+}
+
+func TestWrapper_SignIn(t *testing.T) {
+	err := wrapper.SignIn(cusername, cpassword)
+	if err != nil {
+		t.Errorf("signin failed: %s", err)
+		return
+	}
+}
+
+func TestWrapper_GetUsername(t *testing.T) {
+	un, err := wrapper.GetUsername()
+	if err != nil {
+		t.Errorf("get username failed: %s", err)
+		return
+	}
+	if *un != cusername {
+		t.Errorf("got bad username: %s", un)
```

Comment:
Looks like this doesn't need to be `Errorf`?

Reply:
oh yeah, needs to be `*un`